### DISCUSSION TITLE
Nach DTU Neustart Limit auf 0

### DISCUSSION THREAD

**Discussion by RichieXX** [tbnobody/OpenDTU#772] [General]
Nach Neustart (zB Firmwareupdate) setzt die DTU das Limit des Inverters (HM-1500) immer auf 0. Ich muss dann manuell das Limit wieder auf 100% setzen. Solang der Neustart geplant ist, mag das kein Problem sein, ungeplant schon. Wie kann ich das lösen?

**Comment by DPO99** [comment]
Hast Du das Limit ggf. mal geändert bzw. änderst es regelmäßig über NonPersistant? Ich würde einfach mal 100% mit Persistant rüberbügeln und dann nochmal testen, ob es passt. Das Limit sollte dann immer auf den 100% stehen, wenn er WR neu gestartet wird. Beim Neustart der DTU wird vermutlich das letzte Limit übergeben (aber 0 ist da m.E. nicht erlaubt). Aber Auch da ggf. mal "alle" Limits vorher auf 100% nochmal drüberschicken.

**Comment by RichieXX** [comment]
Ich ändere jedesmal nach Neustart von 0 auf 100% über Persistant. NonPersistant hab ich tatsächlich noch nicht getestet. Der Neustart des WR (was ja jeden Morgen passiert) hat hier keinen Einfluss. Es geht hier nur um den Neustart der DTU. Ich teste beim nächsten Firmwareupdate auch mal NonPersistant und berichte.

**Reply by DPO99** [reply]
0 dürfte eigentlich nicht gehen. Die HM können nur bis 2% runter. 

Persistent ist insofern schlecht, dass du dir damit den Speicher des Wechselrichters auf Dauer kaputt machst.

**Reply by tbnobody** [reply]
Ganz kurz nach dem Start des inverters scheint das Limit auf 0% zu stehen. Das setzt sich der  Inverter aber selbst so. Wenn nun zufällig genau zu dem Zeitpunkt das Limit abgefragt wird ist es 0. Dies setzt sich aber normal beim nächsten Abruf auf den korrekten Wert. Konnte ich hier eben beim Testen auch Nachvollziehen. Lässt sich aber auch nicht verhindern.

**Reply by RichieXX** [reply]
Sorry, war ein paar Tage verreist und konnt nicht antworten.
Habe heute wieder ein Firmwareupdate gemacht und wieder die gleiche Situation. Nun habe ich anstatt des Persist auf Non-Persist das Limit wieder zurücj auf 100% gesetzt, was auch soweit wieder funktioniert hat.
@tbnobody Wird den bei einem Firmwareupdate der DTU der Inverter neu gestartet?

**Comment by tbnobody** [comment]
> @tbnobody Wird den bei einem Firmwareupdate der DTU der Inverter neu gestartet?

Nein. Neustarts werden ausschließlich via WebAPI oder MQTT angestoßen.

Wenn du dich mit einem Tool wie MQTT-Explorer (oder auch auf der CLI mit mosquitto_sub) auf den MQTT Broker verbindest, was ist denn der erste Wert den das CMD topic annimmt? Also ist dort retained ein Wert hinterlegt?

**Comment by RichieXX** [comment]
Ich nutze MQTT mit dem iobroker. Mit dem MQTT Explorer hatte ich die cmd Objekte angelegt. Offensichtlich ohne das retain Flag. Ich habe jetzt mal nachgeholt. Ich melde mich nach dem nächsten Firmwareupdate.

**Reply by DPO99** [reply]
Aha! Das ist schonmal eine wichtige Info, denn der Iobroker MQTT Broker ist etwas zickig mit der OpenDTU, wenn man gewissen Sachen nicht beachtet. Stell mal bitte einen Screenshot deiner MQTT Einstellungen (beide Seiten) hier rein. Ich hatte Anfangs auch das Problem, dass der MQTT Broker von IoBroker nach einem Reconnect des MQTT Gerätes alle möglichen States published. Das muss man erstmal alles einstellen, damit da nix unerwünschtes gesendet wird. Zusätzlich kann der MQTT Broker von Iobroker nicht pro Befehl unterscheiden, ob der Wert mit oder ohne Retain gesendet wird. Das muss man dann über mehrere Client Instanzen im Iobroker lösen...


Ich hab das ganze dann so gelöst, dass ich eine Instanz für den MQTT Broker nutze und eine weitere Instanz als MQTT Client. Über den Client sende ich dann z.B. alles ohne Retain (da non Persistant). Zusätzlich habe ich alle Haken außer (Nur Änderungen senden) entfernt, da dann auch keine unerwünschten Werte bei Restart von IoBroker oder OpenDTU gesendet werden.

Ansonsten fällt mir spontan kein weiterer Grund ein, wieso das Limit einfach auf 0 gesetzt wird und bleibt wenn die DTU neustartet....

**Comment by tbnobody** [comment]
> Mit dem MQTT Explorer hatte ich die cmd Objekte angelegt.

Schaue dir doch bitte mal die Funktionsweise von MQTT an. Man kann dort keine Objekte (Topics) anlegen (auch wenn ioBroker das ggf. so darstellt). Man schickt (published) etwas an die Topic. Der Broker verteilt die Werte an alle Clients die auf die Topic hören (subscribed sind). Danach ist die Topic wieder weg vom Broker. Nichts deutet mehr darauf hin, das sie jemals existiert hätte. Wenn sich also ein neuer Client zum Broker verbindet bekommt er erstmal keine Werte. Er bekommt erst Werte zugeschickt wenn jemand etwas an eine Topic schickt auf die er hört.

Das ist der ganz normale Regelfall. Jetzt gibt es noch das Retain Kennzeichen das man mit einer Nachricht mitschicken kann. Dadurch speichert der Broker die Topic und genau diesen Wert bei sich lokal. Wenn sich ein neuer Client verbindet bekommt er diesen zwischengespeicherten Wert genau dieser Topic zugesandt. 

Wenn weitere Werte ohne Retain Flag gesendet werden, werden diese ganz normal an alle Clients verteilt. Ein sich neu verbindender Client bekommt jedoch den ursprünglichen Wert welcher mit Retain geschickt wurde als initialen Wert.

**Comment by RichieXX** [comment]
@tbnobody 
Danke für deine ausführlichen Erläuterungen. Ehrlich gesagt, habe ich mich tatsächlich noch nicht sehr mit der Funktionsweise von MQTT beschäftigt, weil ich es bis dato nicht musste. Deine Ausführungen bringen mich da ein ganzes Stück weiter. Ich werde mich die nächsten Tage mal reinlesen. Vielleicht finde ich auch dann die Ursache.

**Comment by sturmi82** [comment]
Hallo

wurde das problem bereits gelöst? Denn mein openDTU stellt auch jeden morgen auf 0% und bleibt dann dort stehen.
Ich erinner mich das ganz am anfang mal das die dtu selbst geregelt hat. 
Jetzt mitlerweile stelle ich jeden tag von 0 auf 100% und erst dann regelt er selbst.
Habe die FW 24.1.26 probiert sowie die 23.12.28

**Comment by tbnobody** [comment]
OpenDTU setzt niemals aktiv irgendein Powerlimit. Es wird nichts selbstständig geregelt oder angepasst. Das muss alles von außen kommen. Wenn dein Inverter nach einem Neustart (was auch über Nacht der Fall ist) ein Limit von 0% hat, ist dieses vmtl. irgendwann Persistent gesetzt worden (Siehe WebUI temporär vs. persistent). Hier hilft es vllt. einmal das Limit persistent auf 100% zu setzen.

**Reply by sturmi82** [reply]
Richtig in dem Falle habe ich vergessen zu erwähnen das die DTU von einem Smart Micro Solar geregelt wird. 

Mit persistent meinst du sicherlich 100% Limit auf dauerhaft einstellen

**Comment by wanderfalk** [comment]
Ich habe auch das Problem,  dass das Limit nach einem (Neu-)Start des MSP zunächst ein Limit von 0% anzeigt. Nach einiger Zeit, oder wenn ich eine andere Seite der opendtu anzeigen lasse und dann zurück zur Live-Ansicht gehe, wird korrekte 100% Wert angezeigt. 
Die aktuelle Leistung wird auch vorher schon korrekt (bzw. plausibel, da keine Vergleichsmessung)  angezeigt, obwohl das Limit bei 0% steht
HM-600
v24.6.29

**Reply by tbnobody** [reply]
Bitte ab und zu die Doku lesen...
https://www.opendtu.solar/firmware/faq/
![image](https://github.com/user-attachments/assets/8abca37a-a32a-4331-be5c-26fa924b4188)

**Reply by wanderfalk** [reply]
Diese Diskussion sah so aus, als sei da noch ein offener Bug....
Aber Dankeschön!
